TetrisBot/TetBrain.cs

```diff
@@ -216,13 +215,17 @@ public List<Move> getAllPossibleMoves()
             List<Move> possibleMoves = new List<Move>();
 
             saveState(StateSynced);
-
+            
             for (int rotations = 0; rotations < 4; rotations++)
-            {
+            {//I get this first loop. 4 rotations, 4 possible ways you could rotate a piece.
                 List<int> originalPos = new List<int>();
 
                 for (int translations = -5; translations <= 5; translations++)
-                {
+                {//I don't get this loop. Why 10 translations? surely there are only 3 translations -
+                 //left, right and no move?  Also, why is this a nested loop? If we are looking at all possible
+                 //moves, their options are a rotation and a translation?
+
+                    //There's some interesting and cool code here. If you want help with it, let me know.
                     loadState(prevGameState);
```

Comment:
So, your game grid has an x-axis of 10 points right. As opposed to thinking about a move as either one* move right or left, I thought that 5 moves to the left can be considered as a single move(or potential position) to evaluate. Then 4 moves to the left, then 3 etc. all the way until 5 moves to the right. That way, the outcome of each position in the x-axis is evaluated from left to right. 
Does that make sense, or do you think there is a better way of doing this?

Reply:
Ok, I think I see what you're getting at. Let me think about it a bit this weekend.

If you have a look at the fork on my page, I've added a wiki page with my ramblings, so you can see what I was thinking.

Also, don't forget to accept the pull request and merge :)